src/v/kafka/server/group.cc

```diff
@@ -3559,4 +3559,30 @@ group::delete_expired_offsets(std::chrono::seconds retention_period) {
     return offsets;
 }
 
+std::vector<model::topic_partition>
+group::delete_offsets(std::vector<model::topic_partition> offsets) {
+    std::vector<model::topic_partition> deleted_offsets;
+    /*
+     * Delete the requested offsets, unless the group_state is not empty and
+     * there is at least one active subscription
+     */
+    for (const auto& offset : offsets) {
+        if (in_state(group_state::empty) || !subscribed(offset.topic)) {
+            vlog(_ctxlog.debug, "Deleting group offset {}", offset);
+            _offsets.erase(offset);
+            _pending_offset_commits.erase(offset);
+            deleted_offsets.push_back(offset);
+        }
+    }
```

Comment:
is there another case where the group is neither empty nor dead but also not a consumer group so the error code is that it is a non-empty group?

Reply:
Yup seems like thats about right